lib/db-utils.js

```diff
@@ -173,21 +173,49 @@ module.exports = (srf, logger) => {
         const gw = gwAcc
           .concat(gwSP)
           .sort((a, b) => b.netmask - a.netmask);
-        const selected = gw.find(gatewayMatchesSourceAddress.bind(null, logger, req.source_address));
-        if (selected) {
+        const gateways = gw.filter(gatewayMatchesSourceAddress.bind(null, logger, req.source_address));
+        let voip_carriers = gateways.map((gw) => {
+          return {
+            voip_carrier_sid: gw.voip_carrier_sid,
+            name: gw.name,
+            service_provider_sid: gw.service_provider_sid,
+            account_sid: gw.account_sid,
+            application_sid: gw.application_sid
+          };
+        });
+        /* remove duplicates, winnow down to voip_carriers, not gateways */
+        voip_carriers = [...new Set(voip_carriers.map(JSON.stringify))].map(JSON.parse);
```

Comment:
Could we eliminate the JSON parse and stringify if in fact there is a single carrier (which will be the overwhelming case on most systems)?

Reply:
Applied the change to this line and also to the place it was copied from